src/Image.tsx

```diff
@@ -24,17 +41,16 @@ export default class Image extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      widthDescriptorOnly: Object.keys(this.props.srcSet).every((descriptor) => {
-        return matchWidthDescriptor(descriptor);
-      }),
+      widthDescriptorOnly: Object.keys(this.props.srcSet).every((descriptor) => matchWidthDescriptor(descriptor)),
     };
   }
 
   buildSrcSet() {
     const matcher = this.state.widthDescriptorOnly ? matchWidthDescriptor : matchPixelDescriptor;
     return Object.keys(this.props.srcSet)
       .filter(matcher)
-      .map(descriptor => `${this.props.srcSet[descriptor]} ${descriptor}`);
+      .map(descriptor => `${this.props.srcSet[descriptor]} ${descriptor}`)
+      .join(',');
```

Comment:
Previously this returned array is **implicitly converted to string**.
now the tsc throws error if not `join(',')`ed, because `srcSet::ImgHTMLAttributes<HTMLImageElement>` requires `string | undefined`, not `string[]`.